specification/monitor/resource-manager/microsoft.insights/stable/2018-01-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,430 @@
+                        "retention": "P93D"
+                      }
+                    ],
+                    "dimensions": [{
+                        "value": "Instance",
+                        "localizedValue": "Instance"
+                      }
+                    ]
+                  }, {
+                    "id": "/subscriptions/07c0b09d-9f69-4e6e-8d05-f59f67299cb2/resourceGroups/Rac46PostSwapRG/providers/Microsoft.Web/sites/alertruleTest/providers/microsoft.insights/metricdefinitions/AverageMemoryWorkingSet",
+                    "resourceId": "/subscriptions/07c0b09d-9f69-4e6e-8d05-f59f67299cb2/resourceGroups/Rac46PostSwapRG/providers/Microsoft.Web/sites/alertruleTest",
+                    "namespace": "Microsoft.Web/sites",
+                    "name": {
+                      "value": "AverageMemoryWorkingSet",
+                      "localizedValue": "Average memory working set"
+                    },
+                    "isDimensionRequired": false,
+                    "unit": "Bytes",
+                    "primaryAggregationType": "Average",
+                    "supportedAggregationTypes": [
+                        "None",
+                        "Average",
+                        "Minimum",
+                        "Maximum",
+                        "Total",
+                        "Count"
+                    ],
+                    "metricAvailabilities": [{
+                        "timeGrain": "PT1M",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT5M",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT15M",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT30M",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT1H",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT6H",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "PT12H",
+                        "retention": "P93D"
+                      }, 
+					  {
+                        "timeGrain": "P1D",
+                        "retention": "P93D"
+                      }
+                    ],
+                    "dimensions": [{
+                        "value": "Instance",
+                        "localizedValue": "Instance"
+                      }
+                    ]
+                  }
+                ]
+              } 
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Metric Definitions without filter": { "$ref": "./examples/GetMetricDefinitions.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
```

Comment:
>definitions [](start = 3, length = 11)

Please add `"type": "object"` to the definitions that are type objects

Reply:
Talked to colleague who helped publish the preview swagger and SDK, at this point, we should not change these types as previous version has already been published this way.  And for the change I am making in this version, only include one optional parameter: metricnamespace, which should be string type.